pkg/secrets/secrets.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"context"
+	"crypto/rsa"
+	"fmt"
+	"hash/crc32"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+)
+
+// GetSecret reads a secret from Secret Manager and validates that it was not
+// corrupted during retrieval. The secretResourceName should be in the format:
+// 'projects/*/secrets/*/versions/*'.
+func GetSecret(ctx context.Context, secretResourceName string) (string, error) {
+	sm, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed to create secret manager client: %w", err)
+	}
+	secret, err := ReadSecret(ctx, sm, secretResourceName)
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve secret: %w", err)
+	}
+	err = sm.Close()
+	if err != nil {
+		return "", fmt.Errorf("failed to close secret manager client: %w", err)
+	}
+	return secret, nil
+}
+
+// ReadSecret reads a secret from Secret Manager using the give client and
+// validates that it was not corrupted during retrieval. The secretResourceName
+// should be in the format: 'projects/*/secrets/*/versions/*'.
+func ReadSecret(ctx context.Context, client *secretmanager.Client, secretResourceName string) (string, error) {
```

Comment:
"Read" is not the correct name here, it's "Access". "Read" is the HTTP GET request that gets _metadata_, but not the value.

Reply:
changed name to `AccessSecret`